typescript/opensearch/cwlogs_ingestion/lib/os_setup_stack.ts

```diff
@@ -0,0 +1,244 @@
+              {
+              AllowFromPublic: false,
+              Rules: [
+                  {
+                  ResourceType: 'collection',
+                  Resource: [`collection/${this.COLLECTION_NAME}`],
+                  },
+              ],
+              SourceVPCEs: [vpcEndpoint.attrId],
+              },
+              {
+              AllowFromPublic: true,
+              Rules: [
+                  {
+                  ResourceType: 'dashboard',
+                  Resource: [`collection/${this.COLLECTION_NAME}`],
+                  },
+              ],
+              },
+          ]),
+      });
+
+      // Create OpenSearch Serverless encryption policy
+      const cfnEncryptionPolicy = new CfnSecurityPolicy(this, `${this.STACK_RESOURCE_NAMING_PREFIX}EncryptionPolicy`, {
+            name: this.ENCRYPTION_POLICY_NAME,
+            type: 'encryption',
+            policy: JSON.stringify({
+            Rules: [
+                {
+                ResourceType: 'collection',
+                Resource: [`collection/${this.COLLECTION_NAME}`],
+                }
+            ],
+            AWSOwnedKey: true
+            })
+        });
+
+        // Create OpenSearch Serverless collection
+        const cfnCollection = new CfnCollection(
+            this,
+            `${this.COLLECTION_NAME}Collection`,
+            {
+            name: this.COLLECTION_NAME,
+            description: 'OpenSearch serverless collection to be used for search from CDK',
+            type: 'SEARCH',
+            },
+        );
+
+        cfnCollection.addDependency(cfnEncryptionPolicy);
+        cfnCollection.addDependency(cfnNetworkAccessPolicy);
+
+        // Create IAM role for OpenSearch Ingestion pipeline
+        const pipelineRole = new Role(this, `${this.STACK_RESOURCE_NAMING_PREFIX}PipelineRole`, {
+            roleName: `${this.STACK_RESOURCE_NAMING_PREFIX}PipelineRole`,
+            assumedBy: new ServicePrincipal('osis-pipelines.amazonaws.com'),
+            inlinePolicies: {
+            'OSISPipelineRolePolicy': this.pipelinePolicies(cfnCollection.attrArn)
+            }
+        });   
+
+        // Create OpenSearch Ingestion pipeline        
+        const cfnPipeline = new CfnPipeline(this, `${this.STACK_RESOURCE_NAMING_PREFIX}Pipeline`, {
+            maxUnits: 4,
+            minUnits: 2,
+            bufferOptions: {
+              persistentBufferEnabled: true
+            },
+            pipelineConfigurationBody: this.getPipelineConfiguration(pipelineRole.roleArn, cfnCollection.attrCollectionEndpoint),
+            pipelineName: this.PIPELINE_NAME,
+        });
+
+        this.ingestionEndPointURL = Fn.select(0, cfnPipeline.attrIngestEndpointUrls);
+
+        cfnPipeline.addDependency(cfnCollection);
+
+        // Create a dashboard access role
+        const dashboardAccessRole = new Role(this, `${this.STACK_RESOURCE_NAMING_PREFIX}DashboardAccessRole`, {
+          assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
```

Comment:
Why is this role assumable by EC2? Wouldn't we have the account be the one to assume it?